packages/api/src/graphql/mutation.private.ts

```diff
@@ -737,6 +740,32 @@ export const GraphQLAdminMutation = new GraphQLObjectType<undefined, Context>({
         authorise(CanPublishPage, roles)
         return dbAdapter.page.unpublishPage({id})
       }
+    },
+
+    takeActionOnComment: {
+      type: GraphQLNonNull(GraphQLComment),
+      args: {
+        id: {type: GraphQLNonNull(GraphQLID)},
+        input: {type: GraphQLNonNull(GraphQLCommentActionInput)}
+      },
+      async resolve(root, {id, input}, {authenticate, dbAdapter}) {
+        const {roles} = authenticate()
+        authorise(CanTakeActionOnComment, roles)
+
+        const {state, rejectionReason} = input
+        if (
+          [CommentState.PendingUserChanges, CommentState.Rejected].includes(state) &&
+          !rejectionReason
+        ) {
+          throw new Error('Rejection Reason is required')
```

Comment:
If you throw errors in the GraphQL API they should extend `ApolloError`. Check the file `packages/api/src/error.ts`

Reply:
After changing the mutation to multiple ones, we don't need this check anymore.